src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,265 @@
+                }
+                catch
+                {
+                    return null;
+                }
+            }
+
+            if (uri == null || (uri.Scheme != Uri.UriSchemeHttp && uri.Scheme != Uri.UriSchemeHttps))
+            {
+                return null;
+            }
+            return uri;
+        }
+
+        /// <summary>
+        /// Converts string containing user:password to NetworkCredential object
+        /// </summary>
+        private NetworkCredential GetCredientialsFromString(string value)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                return null;
+            }
+            int idx = value.IndexOf(":");
+            if (idx < 0)
+            {
+                // only user name without password
+                return new NetworkCredential(value, "");
+            }
+            else
+            {
+                return new NetworkCredential(value.Substring(0, idx), value.Substring(idx+1, value.Length - idx -1 ));
+            }
+        }
+
+        /// <summary>
+        /// This function returns true if given Host match bypass list.
+        /// Note, that the list is common for http and https.
+        /// </summary>
+        private bool IsMatchInBypassList(Uri input)
+        {
+            if (_bypass != null)
+            {
+                foreach (string s in _bypass)
+                {
+                    if (s[0] == '.')
+                    {
+                        // This should match either domain it self or any subdomain or host
+                        // .foo.com will match foo.com it self or *.foo.com
+                        if ((s.Length - 1) == input.Host.Length &&
+                            String.Compare(s, 1, input.Host, 0, input.Host.Length, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            return true;
+                        }
+                        else if (input.Host.EndsWith(s, StringComparison.OrdinalIgnoreCase))
+                        {
+                            return true;
+                        }
+
+                    }
+                    else
+                    {
+                        if (String.Equals(s, input.Host, StringComparison.OrdinalIgnoreCase))
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Gets the proxy URI. (iWebProxy interface)
+        /// </summary>
+        public Uri GetProxy(Uri uri)
+        {
+            return uri.Scheme == Uri.UriSchemeHttp ? _http : _https;
+        }
+
+        /// <summary>
+        /// Checks if URI is subject to proxy or not.
+        /// </summary>
+        public bool IsBypassed(Uri uri)
+        {
+            bool ret =  (uri.Scheme == Uri.UriSchemeHttp ? _http : _https) == null;
+
+            if (ret)
+            {
```

Comment:
This can be,

```c#
return  (uri.Scheme == Uri.UriSchemeHttp ? _http : _https) == null ? true : IsMatchInBypassList(uri);
```

Reply:
I'm not sure we even care if the proxy Uri is null here.

Reply:
I did not want to go to the bypass list. I may not understand the comment but the list is common   but we may proxy only for http or https.